src/libUtils/TimeUtils.cpp

```diff
@@ -33,3 +34,16 @@ uint256_t get_time_as_int()
         = duration_cast<microseconds>(system_clock::now().time_since_epoch());
     return static_cast<uint256_t>(microsecs.count());
 }
+
+struct tm* gmtime_safe(const time_t* timer)
+{
+    std::lock_guard<std::mutex> guard(gmtimeMutex);
+    return gmtime(timer);
+}
+
+long int get_ms(const time_point<high_resolution_clock> time)
```

Comment:
Can we pass `time` by reference here, to improve performance?

Reply:
ok